src/mon/commands/Command.h

```diff
@@ -0,0 +1,333 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2019 SUSE LLC <contact@suse.com>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#ifndef CEPH_MON_COMMAND_H
+#define CEPH_MON_COMMAND_H
+
+#include <string>
+#include <set>
+#include <sstream>
+
+#include <boost/intrusive_ptr.hpp>
+
+#include "mon/Monitor.h"
+#include "mon/MonOpRequest.h"
+
+#include "include/types.h"
+#include "include/encoding.h"
+#include "include/ceph_assert.h"
+#include "include/Context.h"
+
+#include "common/cmdparse.h"
+#include "common/RefCountedObj.h"
+
+using namespace std;
+
+/*
+ * CommandCommon
+ *
+ * Define members and functions common to commands.
+ */
+struct Common
+{
+
+  explicit Common(
+      Monitor *_mon,
+      PaxosService *_svc,
+      CephContext *_cct) :
+    mon(_mon),
+    service(_svc),
+    cct(_cct)
+  { }
+
+  virtual ~Common() { }
+
+ private:
+  Common() = delete;
+  void wait_for_finished_proposal(MonOpRequestRef op, Context *ctx) {
+    service->wait_for_finished_proposal(op, ctx);
+  }
+
+ protected:
+  Monitor *mon;
+  PaxosService *service;
+  CephContext *cct;
+
+  version_t get_last_committed() {
```

Comment:
mark it `const`?

Reply:
Since that's how it's declared in the PAXOS service, it makes sense to follow suit:

  version_t get_last_committed() const{
    return cached_last_committed;
  }